File: uploads/docsnsi/comparaison_langage/factorielle.js
const factorielle = 
          n => {
                  if ( n === 0) 
                    return 1;
                  else if ( n === 1) 	
                    return 1;
                  else {
                    let result = 1;
                    let i = 0;
                    while (i < n) {
                      i = i + 1;
                      result = result * i;
                    }
                    return result;
                  }
              }

File: uploads/docsnsi/comparaison_langage/factorielle.pl
factorielle(0,1) :- !.
factorielle(1,1) :- !.
factorielle(N, Result) :- N > 0, 
    					  N_moins_1 is N-1,
    					  factorielle(N_moins_1,   Fact_N_moins_1),
    					  Result is N * Fact_N_moins_1.